taxi/views.py

```diff
@@ -60,17 +62,38 @@
 
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
+    template_name = "taxi/car_detail.html"
+    context_object_name = "car"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        car = self.get_object()
+        context["driver_ids"] = car.drivers.values_list("id", flat=True)
+
+        return context
+
+    def post(self, request, *args, **kwargs):
+        car = self.get_object()
+        if "assign" in request.POST:
+            print(request.POST, request.user)
+            car.drivers.add(request.user)
+        elif "delete" in request.POST:
+            car.drivers.remove(request.user)
+        else:
+            raise PermissionDenied
```

Comment:
Consider adding a message to the `PermissionDenied` exception for better clarity on why the permission was denied.